sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FullOperationResponse, OperationOptions } from "@azure/core-client";
+import { LanguageDetectionInput, TextDocumentInput } from "./generated";
+import { LroResponse } from "@azure/core-lro";
+import { TextAnalysisOperationOptions } from "./models";
+import { logger } from "./logger";
+
+interface IdObject {
+  id: string;
+}
+
+/**
+ * Given a sorted array of input objects (with a unique ID) and an unsorted array of results,
+ * return a sorted array of results.
+ *
+ * @internal
+ * @param sortedArray - An array of entries sorted by `id`
+ * @param unsortedArray - An array of entries that contain `id` but are not sorted
+ */
+export function sortResponseIdObjects<T extends IdObject, U extends IdObject>(
+  sortedArray: T[],
+  unsortedArray: U[]
+): U[] {
+  const unsortedMap = new Map<string, U>();
+  for (const item of unsortedArray) {
+    unsortedMap.set(item.id, item);
+  }
+
+  if (unsortedArray.length !== sortedArray.length) {
+    const ordinal = unsortedArray.length > sortedArray.length ? "more" : "fewer";
+    logger.warning(
+      `The service returned ${ordinal} responses than inputs. Some errors may be treated as fatal.`
+    );
+  }
+
+  const result: U[] = [];
+  for (const sortedItem of sortedArray) {
+    const item = unsortedMap.get(sortedItem.id);
+    if (item) {
```

Comment:
Is the else case an error here? Seems like the service could return the same number of elements as were requested, but not necessarily the same exact set of IDs, in which case you'd have no corresponding result and a really weird error that you can check here easily.

Reply:
Done.